operators/test/e2e/test/apmserver/checks_apm.go

```diff
@@ -47,7 +57,23 @@ func (c *apmClusterChecks) BuildApmServerClient(apm apmtype.ApmServer, k *test.K
 					return err
 				}
 				c.apmClient = apmClient
-				return nil
+
+				// Get the associated Elasticsearch, we assume here that ElasticsearchRef is not nil and that the
+				// Elasticsearch object has been created before the APM Server.
+				var es v1alpha1.Elasticsearch
+				namespace := apm.Spec.ElasticsearchRef.Namespace
+				if len(namespace) == 0 {
```

Comment:
When would we expect that?

Reply:
When the selector contains only the name Elasticsearch object:

```
  elasticsearchRef:
    name: "elasticsearch-sample"
```